src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,22 +42,48 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
+        auto offset = _base_offset() + record.offset_delta();
+
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error(
+              make_error_code(error_code::unable_to_serialize),
+              fmt::format(
+                "Unable to serialize key of record at offset {} in "
+                "topic:partition {}:{}",
+                offset,
+                _tpv.topic(),
+                _tpv.partition()));
+        }
         w.Key("value");
-        rjson_serialize_fmt(_fmt)(w, record.release_value());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_value())) {
+            throw serialize_error(
+              make_error_code(error_code::unable_to_serialize),
+              fmt::format(
+                "Unable to serialize value of record at offset {} in "
```

Comment:
Nit, this text is now spelled out three times. You could create a subclass of `serialize _error` that accepts offset, `tp_view` and wraps all this up.